File: backend/src/main/java/be/kauffman/kfm/vehicule/entity/VehiculeUpdatePayload.java
package be.kauffman.KFM.vehicule.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VehiculeUpdatePayload {
    UUID vehicule_id;
    Date dop;
    Boolean active;
    String price;
    String num_chassis;
    String brand;
    String cde_carrosserie;
    String genre;
    String mom;
    String mma;
    String mmat;
    String mta;
    String classe_enviro;
    String nbr_km;
    String metrologie;
    String picture;
    String fuel;
    String type;
}

File: frontend/src/app/module/vehicule/model/payload/vehicule-update.payload.ts
import {PayloadInterface} from '@shared/model';

export interface VehiculeUpdatePayload extends PayloadInterface {
  dop: Date;
  active: boolean;
  price: string;
  num_chassis: string;
  brand: string;
  cde_carrosserie: string
  genre: string;
  mom: string;
  mma: string;
  mmat: string;
  mta: string;
  classe_enviro: string;
  nbr_km: string;
  metrologie: string;
  picture: string;
  fuel: string;
  type: string;
  vehicule_id: string;
}
